wildfly/extensions/src/main/java/org/keycloak/provider/wildfly/KeycloakSessionServletFilter.java

```diff
@@ -91,28 +94,28 @@ public int getLocalPort() {
             filterChain.doFilter(servletRequest, servletResponse);
         } finally {
             if (servletRequest.isAsyncStarted()) {
-                servletRequest.getAsyncContext().addListener(createAsyncLifeCycleListener(session));
+                servletRequest.getAsyncContext().addListener(createAsyncLifeCycleListener(session, response));
             } else {
-                closeSession(session);
+                closeSession(session, response);
             }
         }
     }
 
-    private AsyncListener createAsyncLifeCycleListener(final KeycloakSession session) {
+    private AsyncListener createAsyncLifeCycleListener(final KeycloakSession session, final HttpServletResponse response) {
```

Comment:
The `response` is available in the `AsyncEvent` object and should not be part of the AsyncListener